bin/extractkernel

```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/perl
+use strict;
+use File::Copy;
+use File::Spec;
+use File::Basename;
+use File::Which;
+use Cwd 'realpath';
+use Getopt::Std;
+use List::Util qw(max);
+
+sub usage {
+  print("Usage: $0 [OPTION]... -i <input>\n");
+  print("Extract the device kernels from an hcc executable.\n\n");
+  print("-h          \t\t\t\tshow this help message\n");
+  print("-i <input>  \t\t\t\tinput file\n");
+  exit;
+}
+
+my $debug = 0;
+
+# use clang offload bundler (instead of "dd")
+# to extract device object from the bundle
+my $use_clang_offload_bundler = 1;
+
+my %options=();
+getopts('hi:', \%options);
+
+if (!%options || defined $options{h}) {
+  usage();
+}
+
+my $input_file;
+defined $options{i} || die("input not specified");
+$input_file = $options{i};
+(-f $input_file) || die("can't find $input_file");
+
+# look for llvm-objdump and clang-offload-bundler
+my $tools_path_prefix;
+my $llvm_objdump;
+my $clang_offload_bundler;
+
+if (defined $ENV{'HCC_HOME'}) {
+  $tools_path_prefix = File::Spec->catfile($ENV{'HCC_HOME'}, "bin");
+  $llvm_objdump = File::Spec->catfile($tools_path_prefix, "llvm-objdump");
+  $clang_offload_bundler = File::Spec->catfile($tools_path_prefix, "clang-offload-bundler");
+}
+else {
+  $tools_path_prefix = dirname(realpath($0));
+  $llvm_objdump = File::Spec->catfile($tools_path_prefix, "llvm-objdump");
+  $clang_offload_bundler = File::Spec->catfile($tools_path_prefix, "clang-offload-bundler");
+  if (!(-f $llvm_objdump)) {
+    $tools_path_prefix = realpath($tools_path_prefix."/../../hcc/bin");
```

Comment:
This should use the `HCC_HOME` path set by cmake. That is this should be set to `$tools_path_prefix = realpath("@HCC_HOME@")` and then use cmake's `configure_file` to set the file.

Reply:
It cannot use a relative path because hcc is a separate package and can be installed in a different location completely unrelated to hip.